examples/chatmemberbot.py

```diff
@@ -136,6 +147,22 @@ async def greet_chat_members(update: Update, context: ContextTypes.DEFAULT_TYPE)
         )
 
 
+async def start_private_chat(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
+    """Greets the user and records that they started a chat with the bot if it's a private chat.
+    Since no `my_chat_member` update is issued when a user starts a private chat with the bot,
+    we have to track it explicitly here, when handling the /start command.
+    """
+    user_name = update.effective_user.full_name
+    chat = update.effective_chat
+    if chat.type == Chat.PRIVATE:
```

Comment:
```suggestion
    if chat.type != Chat.PRIVATE or `chat_id` not in context.bot_data["user_ids"]:
        return
```

Tracking the user and sending the `Welcome {user_name}` message only makes sense if the user is not already tracked :)

Reply:
I think you meant `or "chat_id" in context.bot_data["user_ids"]` (not `not`)?

Reply:
Anyway, if it's the first ever interaction with the bot, `context.bot_data` has no key "user_ids" and it triggers `KeyError`.

Reply:
This works, but it's kinda ugly :)

```py
    if (
        chat.type != Chat.PRIVATE 
        or ("user_ids" in context.bot_data and chat.id in context.bot_data["user_ids"])
    ):
        return
```

Reply:
just use `chat.id in context.bot_data.get("user_ids", [])`

Reply:
bingo. I knew that it was `.get()` but somehow didn't figure out how to word that exactly